pytorch_lightning/callbacks/swa.py

```diff
@@ -0,0 +1,206 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Stochastic Weight Averaging Callback
+====================================
+
+"""
+from typing import Callable, Optional, Union
+
+import torch
+from torch import nn
+
+from pytorch_lightning.callbacks import Callback
+from pytorch_lightning.utilities import _PYTORCH_GREATER_EQUAL_1_6_0, rank_zero_warn
+from pytorch_lightning.utilities.exceptions import MisconfigurationException
+
+if _PYTORCH_GREATER_EQUAL_1_6_0:
+    from torch.optim.swa_utils import AveragedModel, SWALR
+
+    class LightningAveragedModel(AveragedModel):
+
+        def __init__(self, pl_module, *args, **kwargs):
+
+            for k, v in vars(pl_module).items():
+                setattr(self, k, v)
+
+            for fn_name in dir(pl_module):
+                if not fn_name.startswith("__") and fn_name != "update_parameters":
+                    setattr(self, fn_name, getattr(pl_module, fn_name))
+
+            super().__init__(pl_module, *args, **kwargs)
+
+
+class StochasticWeightAveragingCallback(Callback):
```

Comment:
to be a bit consistent with the naming, lets try to not include `Callback` in the name (if possible)
```suggestion
class StochasticWeightAveraging(Callback):
```